builders/integrations/relayers/gelato.md

```diff
@@ -36,278 +36,265 @@ Then, take the following steps:
+    event LogBalance(uint256 indexed balance);
+
+    // this function uses this contract's token balance to send
+    // an _amount of tokens to the _to address
+    function sendToFriend(
+        address _token,
+        address _to,
+        uint256 _amount
+    ) external onlyGelatoRelay {
+        // Payment to Gelato
+        // NOTE: be very careful here!
+        // if you do not use the onlyGelatoRelay modifier,
+        // anyone could encode themselves as the fee collector
+        // in the low-level data and drain tokens from this contract.
+        _transferRelayFee();
+
+        // transfer of ERC-20 tokens
+        SafeERC20.safeTransfer(IERC20(_token), _to, _amount);
+
+        emit LogSendToFriend(_to, _amount);
     }
 
-    function sayHi(address _feeToken) external {
-        string memory message = "Hello World";
 
-        emit Success(_msgSender(), _feeToken, message);
+    // this functions emits the current balance of the wallet contract
+    // in an event that we can check on-chain.
+    function balanceOf() external onlyGelatoRelay {
+        // Payment to Gelato
+        // NOTE: be very careful here!
+        // if you do not use the onlyGelatoRelay modifier,
+        // anyone could encode themselves as the fee collector
+        // in the low-level data and drain tokens from this contract.
+        _transferRelayFee();
+
+        emit LogBalance(address(this).balance);
+
     }
 }
 ```
 
-Next, you'll create a new test account that will submit the gasless transaction. This account is insecure and should not be used in production. This example defines a `test_token` with a default value for demonstration purposes, but you can specify any token here that you'd like.
+`sendToFriend` takes as arguments the address of a friend, the token contract address, and the amount one would like to send. Notice that `sendToFriend` uses the `onlyGelatoRelay` modifier, inherited from `GelatoRelayContext`, which is very important! It means that only the Gelato Relay smart contract can call your function (`0xaBcC9b596420A9E9172FD5938620E265a0f9Df92` on all supported networks!) `_transferRelayFee()` is a function inherited from `GelatoRelayContext` which allows you to pay Gelato directly during the call forward.
 
-```
-  const test_token = "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE";
-  // Create mock wallet
-  const wallet = Wallet.createRandom();
-  const sponsor = await wallet.getAddress();
-  console.log(`Mock PK: ${await wallet._signingKey().privateKey}`);
-  console.log(`Mock wallet address: ${sponsor}`);
-```
+Similarly, `balanceOf` is a very simple function, which takes no arguments, and emits an event with the current balance of the wallet. Let's try calling `balanceOf` on the [`myDummyWallet` contract](https://moonbeam.moonscan.io/address/0xA045eb75e78f4988d42c3cd201365bDD5D76D406) using Gelato Relay SDK!
 
-### Add Request Data {: #add-request-data }
+### `balanceOf` Typescript Example
```

Comment:
I think this should be more descriptive because this title will get displayed at the top of the page on the right hand side, and if a user has no context they have no idea what this section is really about